File: j-LIB-core/src/main/java/io/github/dot166/jlib/game/Cloud.java
package io.github.dot166.jlib.game;

import static io.github.dot166.jlib.game.GameView.PARAMS;
import static io.github.dot166.jlib.game.GameView.frand;
import static io.github.dot166.jlib.game.GameView.irand;

import android.content.Context;

import io.github.dot166.jlib.R;

public class Cloud extends Scenery {
    public Cloud(Context context) {
        super(context);
        setBackgroundResource(frand() < 0.01f ? R.drawable.l_cloud_off : R.drawable.l_cloud);
        getBackground().setAlpha(0x40);
        w = h = irand(PARAMS.CLOUD_SIZE_MIN, PARAMS.CLOUD_SIZE_MAX);
        z = 0;
        v = frand(0.15f, 0.5f);
    }
}

File: j-LIB-core/src/main/java/io/github/dot166/jlib/game/Cloud.kt
package io.github.dot166.jlib.game

import android.content.Context
import io.github.dot166.jlib.R

class Cloud(context: Context?) : Scenery(context) {
    init {
        setBackgroundResource(if (GameView.frand() < 0.01f) R.drawable.l_cloud_off else R.drawable.l_cloud)
        background.alpha = 0x40
        h = GameView.irand(GameView.PARAMS.CLOUD_SIZE_MIN, GameView.PARAMS.CLOUD_SIZE_MAX)
        w = h
        z = 0f
        v = GameView.frand(0.15f, 0.5f)
    }
}
